plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/reader/TrinoReaderContext.java

```diff
@@ -0,0 +1,159 @@
+package io.trino.plugin.hudi.reader;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.hudi.avro.AvroSchemaUtils;
+import org.apache.hudi.common.engine.RecordContext;
+import org.apache.hudi.common.model.HoodieAvroIndexedRecord;
+import org.apache.hudi.common.model.HoodieEmptyRecord;
+import org.apache.hudi.common.model.HoodieKey;
+import org.apache.hudi.common.model.HoodieRecord;
+import org.apache.hudi.common.table.HoodieTableConfig;
+import org.apache.hudi.common.table.read.BufferedRecord;
+import org.apache.hudi.common.util.AvroJavaTypeConverter;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+
+public class TrinoReaderContext
+    extends RecordContext<IndexedRecord>
+{
+
+    public TrinoReaderContext(HoodieTableConfig tableConfig) {
+        super(tableConfig, new AvroJavaTypeConverter());
+    }
+
+    @Override
+    public HoodieRecord<IndexedRecord> constructHoodieRecord(BufferedRecord<IndexedRecord> bufferedRecord, String partitionPath)
+    {
+        if (bufferedRecord.isDelete()) {
+            return new HoodieEmptyRecord<>(
+                    new HoodieKey(bufferedRecord.getRecordKey(), partitionPath),
+                    HoodieRecord.HoodieRecordType.AVRO);
+        }
```

Comment:
What about the ordering value and payload class handling?  Do we have test coverage around updates and deletes with lower and higher ordering values?  The current logic can lead to data loss (because the ordering value is 0 indicating it's commit-time-ordered deletes causing the delete to take effect regardless of the ordering value, in the EVENT_TIME_ORDERING merge mode) if the deletes have lower ordering value.